packages/query-core/src/utils.ts

```diff
@@ -68,9 +68,9 @@ export type QueryTypeFilter = 'all' | 'active' | 'inactive'
 
 export const isServer = typeof window === 'undefined' || 'Deno' in globalThis
 
-export function noop(): undefined {
-  return undefined
-}
+export function noop(): void {}
+
+export const returnUndefined = noop as () => undefined
```

Comment:
would overloads also work?

```
export function noop(): undefined
export function noop(): void
export function noop() {}
```

Reply:
Thanks for the suggestion!🙌 reflected in 454956e23a54defc8ef208df55540d73ddebee0